server/reload.go

```diff
@@ -705,6 +705,46 @@ func (s *Server) Reload() error {
 	if err := s.reloadOptions(curOpts, newOpts); err != nil {
 		return err
 	}
+
+	s.mu.Lock()
+	disconnectClients := []*client{}
+	checkClients := func(kind int, clients map[uint64]*client, set PinnedCertSet) {
+		for _, c := range clients {
+			if (c.kind == kind || (c.kind == CLIENT && c.clientType() == kind)) && !c.matchesPinnedCert(set) {
+				disconnectClients = append(disconnectClients, c)
+			}
+		}
+	}
+	if !reflect.DeepEqual(newOpts.TLSPinnedCerts, curOpts.TLSPinnedCerts) {
+		checkClients(NATS, s.clients, newOpts.TLSPinnedCerts)
```

Comment:
For actual CLIENT connections, I am not sure this helper is really nice because we are going to iterate over all clients for as many "type" that we have (NATS, MQTT and WS, even if we have none of clients being actually MQTT and/or WS).

Reply:
fixed